books/views.py

```diff
@@ -60,8 +60,28 @@ def get(self, request):
 
 
 class BookDetailView(TemplateView):
+    form_class = RequestBookForm
     template_name = "books/book_detail.html"
 
     def get(self, request, pk):
         book = Book.objects.get(pk=pk)
-        return render(request, self.template_name, {"book": book})
+        form = self.form_class()
+        return render(
+            request, self.template_name, {"book": book, "form": form}
+        )
+
+    def post(self, request, pk):
+        book = Book.objects.get(pk=pk)
+        form = self.form_class(request.POST)
+        if form.is_valid():
+            if request.user.is_authenticated:
+                if book.status == "AV":
+                    BookHolder.objects.create(
+                        book=book,
+                        holder=request.user,
+                        date_requested=timezone.now(),
+                    )
+                book.status = form.instance.status
+            book.save()
```

Comment:
Consider putting the save logic into your form.
So you would have something like:

```python
class RequestBookForm(forms.ModelForm):
    def save(holder, ...):    # you can pass holder from your view!
         book = super().save(...)  # this will sort the status save for you
         if book.status == "AV":
             BookHolder.objects.create(book=book, holder=holder, ...)
         return book
```

Then here in view you could have:
```python
if form.is_valid():
    form.save(request.user)
```